src/smif/controller/modelrun.py

```diff
@@ -209,8 +213,14 @@ def solve_model(self, model_run, store):
                         decision_iteration=iteration
                     )
 
-                    model_run.sos_model.simulate(data_handle)
-        return data_handle
+                    sub_job_graph = model_run.sos_model.simulate(data_handle)
+                    connect_to = [node for node in sub_job_graph.nodes if sub_job_graph.in_degree(node) == 0]
+
+                    job_graph = networkx.compose(job_graph, sub_job_graph)
+                    for from_node, to_node in itertools.product(connect_from, connect_to):
+                        job_graph.add_edge(from_node, to_node)
+
+        return job_graph
```

Comment:
The next step in implementing this will be to take account of the decision_loop returned from the decision manager. Job graphs will be submitted to the scheduler for each bundle in the decision_loop, where a bundle can consist of multiple decision iterations and timesteps.  This could be implemented as a co-routine, where decision_loop and the scheduler call one another until an agreed end-point is reached.

As such, I think the job graph should be passed to the scheduler within the decision_loop for now.